### PR TITLE
Cleanup Guid formatting

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/mscorlib/shared/System/Guid.cs [hunk 1]
```diff
@@ -1251,41 +1251,37 @@ public string ToString(string format, IFormatProvider provider)
...
-                    guidString = string.FastAllocateString(68);
+                    guidSize = 68;
                     break;
                 default:
                     throw new FormatException(SR.Format_InvalidGuidFormatSpecification);
             }
 
-            unsafe
-            {
-                fixed (char* guidChars = guidString)
```
Comment (jkotas) [inline src/mscorlib/shared/System/Guid.cs]: This pinning is unnecessary. `string.GetRawStringData()` can be used instead.

File: src/mscorlib/shared/System/Guid.cs [hunk 2]
```diff
@@ -1251,41 +1251,37 @@ public string ToString(string format, IFormatProvider provider)
...
                 case 'B':
                 case 'b':
                 case 'P':
                 case 'p':
-                    guidString = string.FastAllocateString(38);
+                    guidSize = 38;
                     break;
                 case 'X':
                 case 'x':
-                    guidString = string.FastAllocateString(68);
```
Comment (jkotas) [inline src/mscorlib/shared/System/Guid.cs]: It is better (smaller code) to call the method just once after the switch statement.

File: src/mscorlib/shared/System/Guid.cs [hunk 3]
```diff
@@ -1345,73 +1339,66 @@ public bool TryFormat(Span<char> destination, out int charsWritten, string forma
...
+                            *p++ = '-';
+                        p += HexsToChars(p, _f, _g);
+                        p += HexsToChars(p, _h, _i);
+                        p += HexsToChars(p, _j, _k);
                     }
+
+                    if (braces != 0)
+                        *p++ = (char)(braces >> 16);
+
+                    Debug.Assert(p - guidChars == guidSize);
```
Comment (jkotas) [inline src/mscorlib/shared/System/Guid.cs]: I have refactored the code to make it possible to assert that we have written as many characters as expected. This refactoring also made the code smaller and a bit faster.

**Comment by jkotas** [comment]
cc @JeremyKuhne @niustephanie

**Comment by jkotas** [comment]
@dotnet-bot test Windows_NT x64 corefx_baseline

**Comment by danmosemsft** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Build and Test  please (hung)

**Pull request merged**